openmp/libomptarget/plugins-nextgen/common/src/Utils/ELF.cpp

```diff
@@ -261,3 +261,27 @@ utils::elf::getSymbol(const ELFObjectFile<ELF64LE> &ELFObj, StringRef Name) {
 
   return nullptr;
 }
+
+Expected<const void *> utils::elf::getSymbolAddress(
+    const object::ELFObjectFile<object::ELF64LE> &ELFObj,
+    const object::ELF64LE::Sym &Symbol) {
+  const ELFFile<ELF64LE> &ELFFile = ELFObj.getELFFile();
+
+  auto SecOrErr = ELFFile.getSection(Symbol.st_shndx);
+  if (!SecOrErr)
+    return SecOrErr.takeError();
+  const auto &Section = *SecOrErr;
+
+  // A section with SHT_NOBITS occupies no space in the file and has no offset.
+  if (Section->sh_type == ELF::SHT_NOBITS)
+    return createError(
+        "invalid sh_type for symbol lookup, cannot be SHT_NOBITS");
```

Comment:
Why is this en error now? The ELF clearly has the symbol and for things like Size estimation we can use it just fine.

Reply:
I wasn't sure what the original intention was, as far as I could tell it would hit this error because there was a check for `nullptr` right after where we would look this up. That being said, even though we can check size and it exists, it still had no data associated with it so it seems kind of weird. If you want to look up a symbol you can use the other ELF Symbol lookup so that's my understanding.

Reply:
Now you can't call getMetadataFromImage anymore to get the size. I thought that could be useful. Anyway, we can add some new API or extend the other one if we need it.

Reply:
That's freely available with `Sym->st_size`. We could make a nicer looking helper if that's needed.

Reply:
Potentially we could make `getMetadataFromImage` do the `nullptr` handling. I just think that it's correct to have the function that gets the associated data to error if no such data exists.